Python/Product/TestAdapter/testlauncher.py

```diff
@@ -1,17 +1,32 @@
-# Copyright (c) Microsoft Corporation. All rights reserved.
-# Licensed under the MIT License.
-
+# Python Tools for Visual Studio
+# Copyright(c) Microsoft Corporation
+# All rights reserved.
+# 
+# Licensed under the Apache License, Version 2.0 (the License); you may not use
+# this file except in compliance with the License. You may obtain a copy of the
+# License at http://www.apache.org/licenses/LICENSE-2.0
+# 
+# THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+# OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+# IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+# MERCHANTABILITY OR NON-INFRINGEMENT.
+# 
+# See the Apache Version 2.0 License for specific language governing
+# permissions and limitations under the License.
 
 import io
 import os
 import sys
 import traceback
 
 def main():
-    cwd, testRunner, secret, port, debugger_search_path, mixed_mode, coverage_file, test_file, args = parse_argv()
-    sys.path[0] = os.getcwd()
+    cwd, testRunner, secret, port, mixed_mode, coverage_file, test_file, args = parse_argv()
+
+    load_debugger(secret, port, mixed_mode)
```

Comment:
at this point in time it looks like you are assuming that the current working directory is debugpy's parent folder, so you can import debugpy without adding it to the path.

but is that guaranteed?

the code that launches the python process with testlauncher.py passes in the desired cwd as argument (which is parsed above), but the call to start process sets the working directory like this:
```
                using (var proc = ProcessOutput.Run(
                    _projectSettings.InterpreterPath,
                    arguments,
                    _projectSettings.WorkingDirectory,
                    env,
                    visible: false,
                    testRedirector,
                    quoteArgs: true,
                    elevate: false,
                    System.Text.Encoding.UTF8,
                    System.Text.Encoding.UTF8
                )) {
```

Reply:
oh does import debugpy work because THIS script happens to be deployed in the parent folder of debugpy?

Reply:
Yes, it is guaranteed. 

["As initialized upon program startup, the first item of this list, path[0], is the directory containing the script that was used to invoke the Python interpreter."](https://docs.python.org/3/library/sys.html#sys.path)